### DISCUSSION TITLE
AWS SDK for Java 7

### DISCUSSION THREAD

**Discussion by jash0909** [aws/aws-sdk-java#2749] [Q&A] [answered]
Hi,We wanted to use an aws sdk for one of our java 7 application to create new log stream and push logs into it. But looks like that aws sdk is only supported starting from java 8. I have tried using a version of com.amazonaws.aws-java-sdk-logs sdk(which apparently supports java 7) but  AWSLogsClient class is deprecated and is recommended to use AWSLogsClientBuilder class instead. But this AWSLogsClientBuilder class is only available in sdk versions supported for java 8 and above . Could anyone please help us understand if there is any work around and other way to achieve the same for the given usecase .

**Comment by debora-ito** [comment, marked as answer]
AWS SDK for Java v1 supports JDK 7. AWSLogsClientBuilder should work fine in JDK 7. Are you seeing errors?

**Reply by jash0909** [reply]
So, the AWSLogsClientBuilder class is only available in AWS SDK for java V1 versions that support only java 8 and above. However the lower versions of AWS SDK for java V1 that support java 7 don't have the AWSLogsClientBuilder class in the jar. And also the AWSLogsClient class is deprecated.

**Reply by debora-ito** [reply]
The `AWSLogsClient()` constructor is deprecated, not the whole class.

This sample should work fine in JDK 7:
```
AWSLogs logsClient = AWSLogsClient.builder()
                .withRegion(Regions.US_WEST_2)
                .build();
```

Does it work in your case? I'm using Java SDK version `1.12.207`.

**Reply by jash0909** [reply]
I shall try it and let u know. Just to reconfirm , com.amazonaws.aws-java-sdk-logs jar version 1.12.207 works for java 7?

**Reply by debora-ito** [reply]
Yes.
